src/sql/workbench/api/node/mainThreadNotebookDocumentsAndEditors.ts

```diff
@@ -401,6 +416,19 @@ export class MainThreadNotebookDocumentsAndEditors extends Disposable implements
 		return id;
 	}
 
+	private async doRunAllCells(codeCells: ICellModel[], editor: MainThreadNotebookEditor): Promise<boolean> {
+		try {
+			if (codeCells && codeCells.length > 0) {
+				for (let i = 0; i < codeCells.length; i++) {
+					await editor.runCell(codeCells[i]);
```

Comment:
how about runCell fails in the middle, do we still continue or stop? Maybe we should stop

Reply:
Also how about if the cell is in running state?